paddle/fluid/framework/ir/mkldnn/reshape_transpose_matmul_fuse_pass_tester.cc

```diff
@@ -0,0 +1,140 @@
+  b1->SetShape({-1, 128, 768});
+  auto* c1 = layers.reshape2(b1, {0, 0, 12, 64}, with_xshapes);
+  c1->SetShape({-1, 128, 12, 64});
+  auto* d1 = layers.transpose2(c1, {0, 2, 1, 3}, with_xshapes);
+  d1->SetShape({-1, 12, 128, 64});
+  auto* a2 = layers.data("a2", {-1, 128, 768});
+  auto* w2 = layers.data("w2", {768, 768}, true);
+  auto* bias2 = layers.data("bias2", {768}, true);
+  auto* b2 = layers.fc(a2, w2, bias2, 2);
+  b2->SetShape({-1, 128, 768});
+  auto* c2 = layers.reshape2(b2, {0, 0, 12, 64});
+  c2->SetShape({-1, 128, 12, 64});
+  auto* d2 = layers.transpose2(c2, {0, 2, 1, 3});
+  d2->SetShape({-1, 12, 128, 64});
+  layers.matmul(d1, d2);
+
+  std::unique_ptr<ir::Graph> graph(new ir::Graph(layers.main_program()));
+  graph->Set("__param_scope__", CreateParamScope());
+
+  int num_reshape_nodes_before = GetNumOpNodes(graph, "reshape2");
+  int num_transpose_nodes_before = GetNumOpNodes(graph, "transpose2");
+  int total_nodes_before = graph->Nodes().size();
+  VLOG(3) << DebugString(graph);
+
+  auto graph_viz_pass = PassRegistry::Instance().Get("graph_viz_pass");
+  std::string test_name{"test_reshape_transpose_matmul"};
+  if (with_xshapes) test_name += "_one_with_xshapes";
+
+  graph_viz_pass->Set("graph_viz_path",
+                      new std::string(test_name + "_before.dot"));
+  graph.reset(graph_viz_pass->Apply(graph.release()));
+
+  auto pass =
+      PassRegistry::Instance().Get("reshape_transpose_matmul_fuse_pass");
+  graph.reset(pass->Apply(graph.release()));
+
+  graph_viz_pass->Get<std::string>(std::string{"graph_viz_path"}) =
+      test_name + "_after.dot";
+  graph.reset(graph_viz_pass->Apply(graph.release()));
+
+  int num_reshape_nodes_after = GetNumOpNodes(graph, "reshape2");
+  int num_transpose_nodes_after = GetNumOpNodes(graph, "transpose2");
+  int total_nodes_after = graph->Nodes().size();
+  VLOG(3) << DebugString(graph);
+
+  PADDLE_ENFORCE_EQ(num_reshape_nodes_before, 2);
+  PADDLE_ENFORCE_EQ(num_reshape_nodes_after, 0);
+  PADDLE_ENFORCE_EQ(num_transpose_nodes_before, 2);
+  PADDLE_ENFORCE_EQ(num_transpose_nodes_after, 0);
+  int removed = 8;  // 2* reshape, reshape_out, transpose, transpose_out
+  if (with_xshapes) removed += 2;  // transpose_xshape, reshape_xshape
+  PADDLE_ENFORCE_EQ(total_nodes_before - removed, total_nodes_after);
+  auto* matmul_op_desc = GetOpNodes(graph, "matmul").at(0)->Op();
+
+  using size = std::vector<int>;
+  auto check = [&matmul_op_desc](std::string a) {
+    std::string shape_str = "shape_" + a;
+    ASSERT_TRUE(matmul_op_desc->HasAttr(shape_str));
+    bool shape_correct = boost::get<size>(matmul_op_desc->GetAttr(shape_str)) ==
+                         size{0, 0, 12, 64};
+    PADDLE_ENFORCE(shape_correct);
+    std::string axis_str = "axis_" + a;
+    ASSERT_TRUE(matmul_op_desc->HasAttr(axis_str));
+    bool strides_correct =
+        boost::get<size>(matmul_op_desc->GetAttr(axis_str)) == size{0, 2, 1, 3};
+    PADDLE_ENFORCE(strides_correct);
```

Comment:
PADDLE_ENFORCE_NOT_NULL

Reply:
ok I changed it to something different please see new version when I push it.